src/app/icd/server/ICDManager.cpp

```diff
@@ -385,7 +385,14 @@ void ICDManager::SetKeepActiveModeRequirements(KeepActiveFlags flag, bool state)
 {
     assertChipStackLockedByCurrentThread();
 
-    mKeepActiveFlags.Set(flag, state);
+    if (state)
```

Comment:
And here: why this change?

Reply:
Because `Set(flag, state)` actually takes a enum value not a bitmask, so there was an implicit conmversion from into an enum value and the enum value was not valid anymore. It worked because we end up using integers, however types were wrong.

Reply:
Signature of used method was `constexpr BitFlags & Set(FlagsEnum flag, bool isSet)`.

Reply:
Ah, I see.  It's trying to set/clear multiple flags at once, OK.